api/types/userloginstate/user_login_state.go

```diff
@@ -99,6 +100,13 @@ func (u *UserLoginState) CheckAndSetDefaults() error {
 	return nil
 }
 
+// Clone returns a copy of the member.
+func (u *UserLoginState) Clone() *UserLoginState {
+	var copy *UserLoginState
+	utils.StrictObjectToStruct(u, &copy)
```

Comment:
Does `StrictObjectToStruct` export a pointer to a pointer?

Reply:
If we changed this to
```go
	var copy UserLoginState
	utils.StrictObjectToStruct(u, &copy)
	return &copy
```
then `((*UserLoginState)(nil)).Clone()` will be equivalent to `&UserLoginState{}`, right now it returns `(*UserLoginState)(nil)`.

Obviously neither option is in any way sane and we should write or derive some clone function - and `awalterschulze/goderive` keeps distinguishing itself in making the entirely wrong choices once again, since it will unconditionally use unsafe and reflect to clone private fields of types which is absolutely wild - but in the meantime the way `StrictObjectToStruct` is being used is the least worst way.